### PR TITLE
feat: smoketests for serve

### PR DESCRIPTION
**Pull request by evenstensberg**
**What kind of change does this PR introduce?**
Smoketests for devserver

**Did you add tests for your changes?**
Kinda
**If relevant, did you update the documentation?**
n/a
**Summary**

Needs to fix symlinks / close connection before running another tests, other than that okay

**Does this PR introduce a breaking change?**

No
**Other information**

### CODE REVIEW

File: smoketests/dev-server.sh [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        ls ./dev-server/*.smoketest.js | xargs -I{} echo "Running:" {} | tee /dev/tty | cut -d':' -f 2 | xargs -I{} node {}
+        sudo lsof -i :3000 | awk '{print $2}' | grep -v "PID" | xargs -I{} kill {} -9
+        if [ "$?" != "0" ]; then
+            teardown
+        fi
+    done
+    echo "============================ DONE RUNNING $i $iterations times ============================"
+done
+
+teardown
```
Comment (evilebottnawi) [inline smoketests/dev-server.sh:65]: Please avoid using bash, it is unnecessary, you can rewrite it using JavaScript

Comment (evenstensberg) [reply smoketests/dev-server.sh:65]: No

Comment (evilebottnawi) [reply smoketests/dev-server.sh:65]: Why?

Comment (evilebottnawi) [reply smoketests/dev-server.sh:65]: /cc @webpack/cli-team Somebody love `bash`? I think using `JavaScript` everywhere is good idea, windows always have bugs with `bash`, we are potentially creating flaky tests, before we have same tests for webpack-dev-server and it was big pain

Comment (evenstensberg) [reply smoketests/dev-server.sh:65]: I think bash is fine.. The setup and teardown is written in js, that's good enough, don't you think? scripting with bash is made for these tasks

Comment (evilebottnawi) [reply smoketests/dev-server.sh:65]: What is the need for a bash if we have javascript, what is benefits? All I see is reading problems, especially for new developers.

**Review by evilebottnawi — CHANGES_REQUESTED** [review]
Smoke tests are not running on CI, look at output, and it is broken in windows as I written above

File: smoketests/dev-server.sh (hunk 1 above)
Comment (evenstensberg) [reply smoketests/dev-server.sh:65]: Let's use bash now and change to js when we are done with smoketests, yes?

Comment (evilebottnawi) [reply smoketests/dev-server.sh:65]: Why do not rewrite it to JS now before it grows into a big rubbish heap?

**Comment by evenstensberg** [comment]
Only thing left is spawning these proceeses async

File: smoketests/dev-server.sh (hunk 1 above)
Comment (evenstensberg) [reply smoketests/dev-server.sh]: Concurrency is pretty hard with js, could you have a look @evilebottnawi ? I think it's better to handle spawning and cleaning up with bash, gave js a chance but with bash it's easier to control the processes

Comment (evilebottnawi) [reply smoketests/dev-server.sh]: @evenstensberg What's easier? You can describe the problem?

Comment (evenstensberg) [reply smoketests/dev-server.sh]: Every spawn is async, so each script will be done at a different time, in addition these operations in the scripts won't be in order, scripting is easier cause they will enforce concurrency by default

Comment (evilebottnawi) [reply smoketests/dev-server.sh]: you can use async/await, or you can rewrite then on jest and using `--runInBand`

**Comment by snitin315** [comment]
Stale

**Pull request closed without merging**